web/packages/core/src/polyfills.ts

```diff
@@ -47,191 +36,164 @@ function replaceFlashInstances(): void {
         }
         for (const elem of Array.from(embeds)) {
             if (RuffleEmbed.isInterdictable(elem)) {
-                const ruffleObject = RuffleEmbed.fromNativeEmbedElement(elem);
-                elem.replaceWith(ruffleObject);
+                const ruffleEmbed = RuffleEmbed.fromNativeEmbedElement(elem);
+                elem.replaceWith(ruffleEmbed);
             }
         }
     } catch (err) {
         console.error(
-            "Serious error encountered when polyfilling native Flash elements: " +
-                err
+            `Serious error encountered when polyfilling native Flash elements: ${err}`
         );
     }
 }
 
 /**
+ * Inject Ruffle into <iframe> and <frame> elements.
  *
+ * This polyfill isn't fool-proof either: On self-hosted builds, it may
+ * not work due to browsers CORS policy or be loaded too late for some
+ * libraries like SWFObject. These should be less of a problem on the
+ * web extension. This polyfill should, however, do the trick in most
+ * cases, but users should be aware of its natural limits.
  */
-function polyfillStaticContent(): void {
-    replaceFlashInstances();
-}
-
+let iframes: HTMLCollectionOf<HTMLIFrameElement>;
+let frames: HTMLCollectionOf<HTMLFrameElement>;
 /**
  *
  */
-function polyfillDynamicContent(): void {
-    // Listen for changes to the DOM. If nodes are added, re-check for any Flash instances.
-    const observer = new MutationObserver(function (mutationsList) {
-        // If any nodes were added, re-run the polyfill to replace any new instances.
-        const nodesAdded = mutationsList.some(
-            (mutation) => mutation.addedNodes.length > 0
-        );
-        if (nodesAdded) {
-            replaceFlashInstances();
-        }
-    });
+function polyfillFrames(): void {
+    // Create live collections to track embed tags.
+    iframes = iframes || document.getElementsByTagName("iframe");
+    frames = frames || document.getElementsByTagName("frame");
```

Comment:
Can we use `??` here?

Reply:
I think `||` and `??` should be equivalent here. If it's better using `??`, let's do it then.
It's a change that could also be made to the object / embed tags, I think.